### DISCUSSION TITLE
ERROR: Failed building wheel for yarl

### DISCUSSION THREAD

**Discussion by awdhesh-kumar27** [Atri-Labs/atrilabs-engine#460] [Help: Installation & Start] [answered]
Hey team, I am facing issue during installing atri  . Please help in resolving the issue . It is showing below code :
""
C:\Users\DELL\AppData\Roaming\Python\Python311\site-packages\pipenv\vendor\attr\_make.py:876: RuntimeWarning: Running interpreter doesn't sufficiently support code object introspection.  Some features like bare super() or accessing __class__ will not work with slotted classes.
  set_closure_cell(cell, cls)
  error: subprocess-exited-with-error

  × Building wheel for frozenlist (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [35 lines of output]
      *********************
      * Accelerated build *
      *********************
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build\lib.win-amd64-cpython-311
      creating build\lib.win-amd64-cpython-311\frozenlist
      copying frozenlist\__init__.py -> build\lib.win-amd64-cpython-311\frozenlist
      running egg_info
      writing frozenlist.egg-info\PKG-INFO
      writing dependency_links to frozenlist.egg-info\dependency_links.txt
      writing top-level names to frozenlist.egg-info\top_level.txt
      reading manifest file 'frozenlist.egg-info\SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      warning: no previously-included files matching '*.pyc' found anywhere in distribution
      warning: no previously-included files matching '*.pyd' found anywhere in distribution
      warning: no previously-included files matching '*.so' found anywhere in distribution
      warning: no previously-included files matching '*.lib' found anywhere in distribution
      warning: no previously-included files matching '*.dll' found anywhere in distribution
      warning: no previously-included files matching '*.a' found anywhere in distribution
      warning: no previously-included files matching '*.obj' found anywhere in distribution
      warning: no previously-included files found matching 'frozenlist\*.html'
      no previously-included directories found matching 'docs\_build'
      adding license file 'LICENSE'
      writing manifest file 'frozenlist.egg-info\SOURCES.txt'
      copying frozenlist\__init__.pyi -> build\lib.win-amd64-cpython-311\frozenlist
      copying frozenlist\_frozenlist.pyx -> build\lib.win-amd64-cpython-311\frozenlist
      copying frozenlist\py.typed -> build\lib.win-amd64-cpython-311\frozenlist
      warning: build_py: byte-compiling is disabled, skipping.

      running build_ext
      building 'frozenlist._frozenlist' extension
      error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for frozenlist
  error: subprocess-exited-with-error

  × Building wheel for multidict (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [61 lines of output]
      *********************
      * Accelerated build *
      *********************
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build\lib.win-amd64-cpython-311
      creating build\lib.win-amd64-cpython-311\multidict
      copying multidict\_abc.py -> build\lib.win-amd64-cpython-311\multidict
      copying multidict\_compat.py -> build\lib.win-amd64-cpython-311\multidict
      copying multidict\_multidict_base.py -> build\lib.win-amd64-cpython-311\multidict
      copying multidict\_multidict_py.py -> build\lib.win-amd64-cpython-311\multidict
      copying multidict\__init__.py -> build\lib.win-amd64-cpython-311\multidict
      running egg_info
      writing multidict.egg-info\PKG-INFO
      writing dependency_links to multidict.egg-info\dependency_links.txt
      writing top-level names to multidict.egg-info\top_level.txt
      reading manifest file 'multidict.egg-info\SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      warning: no previously-included files matching '*.pyc' found anywhere in distribution
      warning: no previously-included files found matching 'multidict\_multidict.html'
      warning: no previously-included files found matching 'multidict\*.so'
      warning: no previously-included files found matching 'multidict\*.pyd'
      warning: no previously-included files found matching 'multidict\*.pyd'
      no previously-included directories found matching 'docs\_build'
      adding license file 'LICENSE'
      writing manifest file 'multidict.egg-info\SOURCES.txt'
      C:\Users\DELL\AppData\Local\Temp\pip-build-env-wd9cn3ed\overlay\Lib\site-packages\setuptools\command\build_py.py:202: SetuptoolsDeprecationWarning:     Installing 'multidict._multilib' as data is deprecated, please list it in `packages`.
          !!

          ############################
          # Package would be ignored #
          ############################
          Python recognizes 'multidict._multilib' as an importable package,
          but it is not listed in the `packages` configuration of setuptools.

          'multidict._multilib' has been automatically added to the distribution only
          because it may contain data files, but this behavior is likely to change
          in future versions of setuptools (and therefore is considered deprecated).

          Please make sure that 'multidict._multilib' is included as a package by using
          the `packages` configuration field or the proper discovery methods
          (for example by using `find_namespace_packages(...)`/`find_namespace:`
          instead of `find_packages(...)`/`find:`).

          You can read more about "package discovery" and "data files" on setuptools
          documentation page.

      !!

        check.warn(importable)
      copying multidict\__init__.pyi -> build\lib.win-amd64-cpython-311\multidict
      copying multidict\py.typed -> build\lib.win-amd64-cpython-311\multidict
      warning: build_py: byte-compiling is disabled, skipping.

      running build_ext
      building 'multidict._multidict' extension
      error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for multidict
  error: subprocess-exited-with-error

  × Building wheel for yarl (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [39 lines of output]
      C:\Users\DELL\AppData\Local\Temp\pip-build-env-tfuvgzdc\overlay\Lib\site-packages\setuptools\config\setupcfg.py:508: SetuptoolsDeprecationWarning: The license_file parameter is deprecated, use license_files instead.
        warnings.warn(msg, warning_class)
      **********************
      * Accelerated build *
      **********************
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build\lib.win-amd64-cpython-311
      creating build\lib.win-amd64-cpython-311\yarl
      copying yarl\_quoting.py -> build\lib.win-amd64-cpython-311\yarl
      copying yarl\_quoting_py.py -> build\lib.win-amd64-cpython-311\yarl
      copying yarl\_url.py -> build\lib.win-amd64-cpython-311\yarl
      copying yarl\__init__.py -> build\lib.win-amd64-cpython-311\yarl
      running egg_info
      writing yarl.egg-info\PKG-INFO
      writing dependency_links to yarl.egg-info\dependency_links.txt
      writing requirements to yarl.egg-info\requires.txt
      writing top-level names to yarl.egg-info\top_level.txt
      reading manifest file 'yarl.egg-info\SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      warning: no previously-included files matching '*.pyc' found anywhere in distribution
      warning: no previously-included files matching '*.cache' found anywhere in distribution
      warning: no previously-included files found matching 'yarl\*.html'
      warning: no previously-included files found matching 'yarl\*.so'
      warning: no previously-included files found matching 'yarl\*.pyd'
      no previously-included directories found matching 'docs\_build'
      adding license file 'LICENSE'
      writing manifest file 'yarl.egg-info\SOURCES.txt'
      copying yarl\__init__.pyi -> build\lib.win-amd64-cpython-311\yarl
      copying yarl\_quoting_c.pyi -> build\lib.win-amd64-cpython-311\yarl
      copying yarl\_quoting_c.pyx -> build\lib.win-amd64-cpython-311\yarl
      copying yarl\py.typed -> build\lib.win-amd64-cpython-311\yarl
      warning: build_py: byte-compiling is disabled, skipping.

      running build_ext
      building 'yarl._quoting_c' extension
      error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for yarl
ERROR: Could not build wheels for frozenlist, multidict, yarl, which is required to install pyproject.toml-based projects
[=== ]
Installation Failed

C:\my_app>  "

**Comment by darshitac11** [comment, marked as answer]
As suggested in the Slack community, there seems to be a problem with the python installed on your computer. 

You should re-install python 3.10.x

**Comment by darshitac11** [comment]
@all-contributors please add @awdhesh-kumar27 as a contributor for bug

**Comment by md-yunus-tabrez** [comment]
I am facing the same issue. Have tried many times & also after re-installing the python, it's showing same error. 

C:\Users\myt07\my_app>pipenv install atri
Installing atri...
Error:  An error occurred while installing atri!
Error text: Collecting atri
  Using cached atri-0.0.83-py3-none-any.whl (22 kB)
Collecting click
  Using cached click-8.1.3-py3-none-any.whl (96 kB)
Collecting python-socketio[asyncio_client]
  Using cached python_socketio-5.7.2-py3-none-any.whl (56 kB)
Collecting toml
  Using cached toml-0.10.2-py2.py3-none-any.whl (16 kB)
Collecting questionary
  Using cached questionary-1.10.0-py3-none-any.whl (31 kB)
Collecting requests
  Using cached requests-2.28.1-py3-none-any.whl (62 kB)
Collecting pymongo[srv]
  Using cached pymongo-4.3.2-cp311-cp311-win_amd64.whl (381 kB)
Collecting colorama
  Using cached colorama-0.4.6-py2.py3-none-any.whl (25 kB)
Collecting dnspython<3.0.0,>=1.16.0
  Using cached dnspython-2.2.1-py3-none-any.whl (269 kB)
Collecting bidict>=0.21.0
  Using cached bidict-0.22.0-py3-none-any.whl (36 kB)
Collecting python-engineio>=4.3.0
  Using cached python_engineio-4.3.4-py3-none-any.whl (52 kB)
Collecting aiohttp>=3.4
  Using cached aiohttp-3.8.3-cp311-cp311-win_amd64.whl (317 kB)
Collecting prompt_toolkit<4.0,>=2.0
  Using cached prompt_toolkit-3.0.32-py3-none-any.whl (382 kB)
Collecting charset-normalizer<3,>=2
  Using cached charset_normalizer-2.1.1-py3-none-any.whl (39 kB)
Collecting idna<4,>=2.5
  Using cached idna-3.4-py3-none-any.whl (61 kB)
Collecting urllib3<1.27,>=1.21.1
  Using cached urllib3-1.26.12-py2.py3-none-any.whl (140 kB)
Collecting certifi>=2017.4.17
  Using cached certifi-2022.9.24-py3-none-any.whl (161 kB)
Collecting attrs>=17.3.0
  Using cached attrs-22.1.0-py2.py3-none-any.whl (58 kB)
Collecting multidict<7.0,>=4.5
  Using cached multidict-6.0.2.tar.gz (50 kB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
Collecting async-timeout<5.0,>=4.0.0a3
  Using cached async_timeout-4.0.2-py3-none-any.whl (5.8 kB)
Collecting yarl<2.0,>=1.0
  Using cached yarl-1.8.1.tar.gz (172 kB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
Collecting frozenlist>=1.1.1
  Using cached frozenlist-1.3.3-cp311-cp311-win_amd64.whl (32 kB)
Collecting aiosignal>=1.1.2
  Using cached aiosignal-1.3.1-py3-none-any.whl (7.6 kB)
Collecting wcwidth
  Using cached wcwidth-0.2.5-py2.py3-none-any.whl (30 kB)
Building wheels for collected packages: multidict, yarl
  Building wheel for multidict (pyproject.toml): started
  Building wheel for multidict (pyproject.toml): finished with status 'error'
  Building wheel for yarl (pyproject.toml): started
  Building wheel for yarl (pyproject.toml): finished with status 'error'
Failed to build multidict yarl

  error: subprocess-exited-with-error

  × Building wheel for multidict (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [67 lines of output]
      *********************
      * Accelerated build *
      *********************
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build\lib.win-amd64-cpython-311
      creating build\lib.win-amd64-cpython-311\multidict
      copying multidict\_abc.py -> build\lib.win-amd64-cpython-311\multidict
      copying multidict\_compat.py -> build\lib.win-amd64-cpython-311\multidict
      copying multidict\_multidict_base.py -> build\lib.win-amd64-cpython-311\multidict
      copying multidict\_multidict_py.py -> build\lib.win-amd64-cpython-311\multidict
      copying multidict\__init__.py -> build\lib.win-amd64-cpython-311\multidict
      running egg_info
      writing multidict.egg-info\PKG-INFO
      writing dependency_links to multidict.egg-info\dependency_links.txt
      writing top-level names to multidict.egg-info\top_level.txt
      reading manifest file 'multidict.egg-info\SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      warning: no previously-included files matching '*.pyc' found anywhere in distribution
      warning: no previously-included files found matching 'multidict\_multidict.html'
      warning: no previously-included files found matching 'multidict\*.so'
      warning: no previously-included files found matching 'multidict\*.pyd'
      warning: no previously-included files found matching 'multidict\*.pyd'
      no previously-included directories found matching 'docs\_build'
      adding license file 'LICENSE'
      writing manifest file 'multidict.egg-info\SOURCES.txt'
      C:\Users\myt07\AppData\Local\Temp\pip-build-env-_7c__6m6\overlay\Lib\site-packages\setuptools\command\build_py.py:202: SetuptoolsDeprecationWarning:     Installing 'multidict._multilib' as data is deprecated, please list it in `packages`.
          !!


          ############################
          # Package would be ignored #
          ############################
          Python recognizes 'multidict._multilib' as an importable package,
          but it is not listed in the `packages` configuration of setuptools.

          'multidict._multilib' has been automatically added to the distribution only
          because it may contain data files, but this behavior is likely to change
          in future versions of setuptools (and therefore is considered deprecated).

          Please make sure that 'multidict._multilib' is included as a package by using
          the `packages` configuration field or the proper discovery methods
          (for example by using `find_namespace_packages(...)`/`find_namespace:`
          instead of `find_packages(...)`/`find:`).

          You can read more about "package discovery" and "data files" on setuptools
          documentation page.


      !!

        check.warn(importable)
      copying multidict\__init__.pyi -> build\lib.win-amd64-cpython-311\multidict
      copying multidict\py.typed -> build\lib.win-amd64-cpython-311\multidict
      warning: build_py: byte-compiling is disabled, skipping.

      running build_ext
      building 'multidict._multidict' extension
      creating build\temp.win-amd64-cpython-311
      creating build\temp.win-amd64-cpython-311\Release
      creating build\temp.win-amd64-cpython-311\Release\multidict
      "C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30133\bin\HostX86\x64\cl.exe" /c /nologo /O2 /W3 /GL /DNDEBUG /MD -IC:\Users\myt07\.virtualenvs\my_app-4nPuCTyO\include -IC:\Users\myt07\AppData\Local\Programs\Python\Python311\include -IC:\Users\myt07\AppData\Local\Programs\Python\Python311\Include "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30133\include" /Tcmultidict/_multidict.c /Fobuild\temp.win-amd64-cpython-311\Release\multidict/_multidict.obj -O2
      _multidict.c
      C:\Users\myt07\AppData\Local\Programs\Python\Python311\include\pyconfig.h(59): fatal error C1083: Cannot open include file: 'io.h': No such file or directory
      error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.29.30133\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for multidict
  error: subprocess-exited-with-error

  × Building wheel for yarl (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [45 lines of output]
      C:\Users\myt07\AppData\Local\Temp\pip-build-env-a0xau4v3\overlay\Lib\site-packages\setuptools\config\setupcfg.py:508: SetuptoolsDeprecationWarning: The license_file parameter is deprecated, use license_files instead.
        warnings.warn(msg, warning_class)
      **********************
      * Accelerated build *
      **********************
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build\lib.win-amd64-cpython-311
      creating build\lib.win-amd64-cpython-311\yarl
      copying yarl\_quoting.py -> build\lib.win-amd64-cpython-311\yarl
      copying yarl\_quoting_py.py -> build\lib.win-amd64-cpython-311\yarl
      copying yarl\_url.py -> build\lib.win-amd64-cpython-311\yarl
      copying yarl\__init__.py -> build\lib.win-amd64-cpython-311\yarl
      running egg_info
      writing yarl.egg-info\PKG-INFO
      writing dependency_links to yarl.egg-info\dependency_links.txt
      writing requirements to yarl.egg-info\requires.txt
      writing top-level names to yarl.egg-info\top_level.txt
      reading manifest file 'yarl.egg-info\SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      warning: no previously-included files matching '*.pyc' found anywhere in distribution
      warning: no previously-included files matching '*.cache' found anywhere in distribution
      warning: no previously-included files found matching 'yarl\*.html'
      warning: no previously-included files found matching 'yarl\*.so'
      warning: no previously-included files found matching 'yarl\*.pyd'
      no previously-included directories found matching 'docs\_build'
      adding license file 'LICENSE'
      writing manifest file 'yarl.egg-info\SOURCES.txt'
      copying yarl\__init__.pyi -> build\lib.win-amd64-cpython-311\yarl
      copying yarl\_quoting_c.pyi -> build\lib.win-amd64-cpython-311\yarl
      copying yarl\_quoting_c.pyx -> build\lib.win-amd64-cpython-311\yarl
      copying yarl\py.typed -> build\lib.win-amd64-cpython-311\yarl
      warning: build_py: byte-compiling is disabled, skipping.

      running build_ext
      building 'yarl._quoting_c' extension
      creating build\temp.win-amd64-cpython-311
      creating build\temp.win-amd64-cpython-311\Release
      creating build\temp.win-amd64-cpython-311\Release\yarl
      "C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30133\bin\HostX86\x64\cl.exe" /c /nologo /O2 /W3 /GL /DNDEBUG /MD -IC:\Users\myt07\.virtualenvs\my_app-4nPuCTyO\include -IC:\Users\myt07\AppData\Local\Programs\Python\Python311\include -IC:\Users\myt07\AppData\Local\Programs\Python\Python311\Include "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30133\include" /Tcyarl/_quoting_c.c /Fobuild\temp.win-amd64-cpython-311\Release\yarl/_quoting_c.obj
      _quoting_c.c
      C:\Users\myt07\AppData\Local\Programs\Python\Python311\include\pyconfig.h(59): fatal error C1083: Cannot open include file: 'io.h': No such file or directory
      error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.29.30133\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for yarl
ERROR: Could not build wheels for multidict, yarl, which is required to install pyproject.toml-based projects
[==  ]
Installation Failed

**Reply by darshitac11** [reply]
Which Python version are you using?

**Reply by md-yunus-tabrez** [reply]
latest version: Python 3.11.0

**Reply by darshitac11** [reply]
There are lots of problems with the recently released 3.11.0 version of Python. For all Python projects, use 3.10.x wherever possible.

**Reply by md-yunus-tabrez** [reply]
Okay, got it.

**Reply by md-yunus-tabrez** [reply]
Yeah! My problem is now solved:

C:\Users\myt07\my_app>pipenv install atri
Installing atri...
Adding atri to Pipfile's [packages]...
Installation Succeeded
Pipfile.lock not found, creating...
Locking [packages] dependencies...
           Building requirements...
Resolving dependencies...
Success!
Locking [dev-packages] dependencies...
Updated Pipfile.lock (b18d7efa6d0a2242c4650434e01b5cd4325df725d796d5f82d3b9f2d790f2ce7)!
Installing dependencies from Pipfile.lock (0f2ce7)...
To activate this project's virtualenv, run pipenv shell.
Alternatively, run a command inside the virtualenv with pipenv run.

i) By upgrading pip 
`pip install --upgrade pip` 

ii) By installing discord 
`pip install discord.py` 

iii) install or upgrade the Setuptools Python package
`pip install --upgrade setuptools`

iv) And by Installation of Microsoft C++ Build Tools Standalone MSVC compiler, libraries, and scripts
Build Tools: `https://visualstudio.microsoft.com/visual-cpp-build-tools/`
